firmware/thruna/BMS/Src/Io/Io_LTC6813/Io_LTC6813CellVoltages.c

```diff
@@ -157,3 +186,185 @@ float Io_LTC6813CellVoltages_GetCellVoltage(uint8_t segement, uint8_t cell)
+
+        if (recv_pec15 == calc_pec15)
+        {
+            // Ignore cells 17 and 18 if reading back from group F. Only read
+            // single word. Otherwise read all bytes stored in rx_buffer
+            const uint8_t num_bytes_to_copy =
+                (curr_reg_group == CELL_V_REG_GROUP_F) ? sizeof(*rx_buffer) : NUM_REG_GROUP_PAYLOAD_BYTES;
+
+            // Store register group data into a temporary array reg_group_data
+            uint16_t reg_group_data[NUM_OF_READINGS_PER_REG_GROUP] = { 0 };
+            memcpy((uint8_t *)reg_group_data, (uint8_t *)&rx_buffer[start_index], num_bytes_to_copy);
+
+            // Write out voltage readings
+            const float reading_0_voltage = CONVERT_100UV_TO_VOLTAGE(reg_group_data[REG_GROUP_READING_0]);
+            const float reading_1_voltage = CONVERT_100UV_TO_VOLTAGE(reg_group_data[REG_GROUP_READING_1]);
+            const float reading_2_voltage = CONVERT_100UV_TO_VOLTAGE(reg_group_data[REG_GROUP_READING_2]);
+
+            const int reading_0_cell_idx = curr_reg_group * NUM_OF_READINGS_PER_REG_GROUP + REG_GROUP_READING_0;
+            const int reading_1_cell_idx = curr_reg_group * NUM_OF_READINGS_PER_REG_GROUP + REG_GROUP_READING_1;
+            const int reading_2_cell_idx = curr_reg_group * NUM_OF_READINGS_PER_REG_GROUP + REG_GROUP_READING_2;
+
+            if (pull_up)
+            {
+                if (reading_0_cell_idx < (int)ACCUMULATOR_NUM_SERIES_CELLS_PER_SEGMENT)
+                {
+                    OWC_cell_V_PU[curr_segment][reading_0_cell_idx] = reading_0_voltage;
+                }
+
+                if (reading_1_cell_idx < (int)ACCUMULATOR_NUM_SERIES_CELLS_PER_SEGMENT)
+                {
+                    OWC_cell_V_PU[curr_segment][reading_1_cell_idx] = reading_1_voltage;
+                }
+
+                if (reading_2_cell_idx < (int)ACCUMULATOR_NUM_SERIES_CELLS_PER_SEGMENT)
+                {
+                    OWC_cell_V_PU[curr_segment][reading_2_cell_idx] = reading_2_voltage;
+                }
+            }
+            else
+            {
+                if (reading_0_cell_idx < (int)ACCUMULATOR_NUM_SERIES_CELLS_PER_SEGMENT)
+                {
+                    OWC_cell_V_PD[curr_segment][reading_0_cell_idx] = reading_0_voltage;
+                }
+
+                if (reading_1_cell_idx < (int)ACCUMULATOR_NUM_SERIES_CELLS_PER_SEGMENT)
+                {
+                    OWC_cell_V_PD[curr_segment][reading_1_cell_idx] = reading_1_voltage;
+                }
+
+                if (reading_2_cell_idx < (int)ACCUMULATOR_NUM_SERIES_CELLS_PER_SEGMENT)
+                {
+                    OWC_cell_V_PD[curr_segment][reading_2_cell_idx] = reading_2_voltage;
+                }
+            }
+        }
+        else
+        {
+            status = false;
+        }
+    }
+
+    return status;
+}
+
+bool Io_LTC6813CellVoltages_Start_OWC(bool pull_up)
```

Comment:
Nit: extra underscore in function names, eg:
```suggestion
bool Io_LTC6813CellVoltages_StartOWC(bool pull_up)
```